### ISSUE TITLE
Rethinkdb causes a panic error on daprd

### ISSUE THREAD

**Issue by bencz** [dapr/components-contrib#1993]
When trying to run a test using Rethinkdb as the state-store, I got the following error:

```
panic: interface conversion: *rethinkdb.RethinkDB is not state.TransactionalStore: missing method Multi

goroutine 1 [running]:
github.com/dapr/dapr/pkg/grpc.NewAPI({0x7ffdce521ea7, 0x12}, {0x0?, 0x0}, {0x4b1e7d8?, 0x734e7d8}, 0xc000623770, 0xc000623740, 0xc000623890, 0xc0006238c0, ...)
        /home/runner/work/dapr/dapr/pkg/grpc/api.go:288 +0x4fb
github.com/dapr/dapr/pkg/runtime.(*DaprRuntime).getGRPCAPI(0xc000ca1b00)
        /home/runner/work/dapr/dapr/pkg/runtime/runtime.go:1176 +0x2fd
github.com/dapr/dapr/pkg/runtime.(*DaprRuntime).initRuntime(0xc000ca1b00, 0xc000d96380)
        /home/runner/work/dapr/dapr/pkg/runtime/runtime.go:417 +0x8b2
github.com/dapr/dapr/pkg/runtime.(*DaprRuntime).Run(0xc000ca1b00, {0xc000b430c8, 0x9, 0xc000b42f40?})
        /home/runner/work/dapr/dapr/pkg/runtime/runtime.go:288 +0x20e
main.main()
        /home/runner/work/dapr/dapr/cmd/daprd/main.go:191 +0x3051
```

## Steps to Reproduce the Problem

Download the attached project
Run docker-compose (watch out as docker-compose has an override file)
Check the sidecar error

[DaprMongoDbBug.zip](https://github.com/dapr/components-contrib/files/9396742/DaprMongoDbBug.zip)

**Comment by berndverst** [comment]
@bencz the issue will be addressed in release 1.9. https://github.com/dapr/components-contrib/pull/1930

I just took a look at the RethinkDB component and have determined that the Etag capability is not supported by RethinkDB. As such this component cannot be used as actor state store. I see you are trying to use it this way, so I wanted to let you know! I will be updating the docs to remove this.

I have enabled state store conformance tests for this component, making this component Beta going forward. But to reiterate, it will not support Etag, and hence cannot support the actor state store.

https://github.com/dapr/components-contrib/pull/2015

**Comment by bencz** [comment]
@berndverst 

RethinkDB works a little differently... there is no transaction in it because it doesn't need to... as queries or upserts are processed by the db in an atomic way, just adjust the update/insert query to validate the ETag field

Check this: https://rethinkdb.com/docs/consistency/
And this: https://stackoverflow.com/questions/40364666/proper-upsert-atomic-update-counter-field-or-insert-document-with-rethinkdb
